sources/engine/Stride.Graphics/Direct3D/SwapChainGraphicsPresenter.Direct3D.cs

```diff
@@ -426,21 +465,26 @@ private SwapChain CreateSwapChainForDesktop(IntPtr handle)
                 newSwapChain.IsFullScreen = true;
 
                 // This is really important to call ResizeBuffers AFTER switching to IsFullScreen 
-                newSwapChain.ResizeBuffers(bufferCount, Description.BackBufferWidth, Description.BackBufferHeight, newFormat: default, SwapChainFlags.AllowModeSwitch);
+                newSwapChain.ResizeBuffers(bufferCount, Description.BackBufferWidth, Description.BackBufferHeight, newFormat: default, description.Flags);
             }
 
             return newSwapChain;
+        }
 
-            bool IsFlipModelSupported()
-            {
-                // From https://github.com/walbourn/directx-vs-templates/blob/main/d3d11game_win32_dr/DeviceResources.cpp#L138
-                using var dxgiDevice = GraphicsDevice.NativeDevice.QueryInterface<SharpDX.DXGI.Device>();
-                using var dxgiAdapter = dxgiDevice.Adapter;
-                using var dxgiFactory = dxgiAdapter.GetParent<SharpDX.DXGI.Factory4>();
-                return dxgiFactory != null;
-            }
+        private SwapChainFlags GetSwapChainFlags()
+        {
+            var flags = SwapChainFlags.None;
+            if (Description.IsFullScreen)
+                flags |= SwapChainFlags.AllowModeSwitch;
+
+            // From https://learn.microsoft.com/en-us/windows/win32/direct3ddxgi/variable-refresh-rate-displays
+            // It is recommended to always use the tearing flag when it is supported.
+            if (tearingSupport)
```

Comment:
Should you check Is Fullscreen here as well (based on the comment on tearing above)?